### PR TITLE
Adds Dynamic Cards support in the dashboard/cards-data Endpoint Response

### LINKED ISSUE #19734: [Dynamic Cards] Update Endpoint Response
### Endpoint
`sites/{SITE_ID}/dashboard/cards-data`

### The Full Response: 
```
{
  "todays_stats": {...},
  "posts": {...},
  "pages": [...],
  "activity": {...},
  "dynamic": [Card, Card, ...]
}
```

### The Card Object:
```
{
  "id": "year_in_review_2023",
  "title": "News",
  "remote_feature_flag": "dynamic_dashboard_cards",
  "featured_image": "https://path/to/image",
  "url": "https://wordpress.com",
  "action": "Call to action",
  "order": "top",
  "rows": [
    {
      "icon": "https://path/to/image",
      "title": "xyz",
      "description": "xyz"
    },
    {
      "icon": "https://path/to/image",
      "title": "xyz",
      "description": "xyz"
    }
  ]
}
```
order will have 2 possible values: “top” and “bottom”.

### PR DESCRIPTION
**Pull request by antonis**
Fixes https://github.com/wordpress-mobile/WordPress-Android/issues/19734

**`WordPress-Android` PR:** https://github.com/wordpress-mobile/WordPress-Android/pull/19773

## Description

This PR adds Dynamic Cards support in the dashboard/cards-data Endpoint Response 

## To Test

Use https://github.com/wordpress-mobile/WordPress-Android/pull/19773

### CODE REVIEW

**Comment by antonis** [comment]
Adding the `Not Ready for Merge` label till we test with the backend

File: fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/dashboard/CardsRestClient.kt [hunk 1]
```diff
@@ -163,6 +173,40 @@ class CardsRestClient @Inject constructor(
...
 
+    data class DynamicCardResponse(
+        @SerializedName("id") val id: String,
+        @SerializedName("title") val title: String,
+        @SerializedName("remote_feature_flag") val remoteFeatureFlag: String,
+        @SerializedName("featured_image") val featuredImage: String,
+        @SerializedName("url") val url: String,
+        @SerializedName("action") val action: String,
+        @SerializedName("order") val order: String,
+        @SerializedName("rows") val rows: List<DynamicCardRowResponse>
```
Comment (justtwago) [inline fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/dashboard/CardsRestClient.kt]: According to the spec (pc8HXX-1nA-p2) we expect some of the properties are nullable:
```suggestion
        @SerializedName("id") val id: String,
        @SerializedName("title") val title: String?,
        @SerializedName("remote_feature_flag") val remoteFeatureFlag: String?,
        @SerializedName("featured_image") val featuredImage: String?,
        @SerializedName("url") val url: String?,
        @SerializedName("action") val action: String?,
        @SerializedName("order") val order: String,
        @SerializedName("rows") val rows: List<DynamicCardRowResponse>
```

**Review by justtwago — COMMENTED** [review]
Thank you for this change and the extended tests! 🚀 
I left a comment about the nullability of some fields.

File: fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/dashboard/CardsRestClient.kt [hunk 2]
```diff
@@ -163,6 +173,40 @@ class CardsRestClient @Inject constructor(
...
+            action = action,
+            order = CardOrder.fromString(order),
+            rows = rows.map { it.toDynamicCardRow() }
+        )
+    }
+
+    data class DynamicCardRowResponse(
+        @SerializedName("icon") val icon: String,
+        @SerializedName("title") val title: String,
+        @SerializedName("description") val description: String
```
Comment (justtwago) [inline fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/dashboard/CardsRestClient.kt]: The same with nullability here. According to the designs we can have different modifications of rows: without an icon, title, or perhaps description. 
Reference: YSMw2nbFAgPpjZLLbYvTfT-fi-1281_9315
```suggestion
        @SerializedName("icon") val icon: String?,
        @SerializedName("title") val title: String?,
        @SerializedName("description") val description: String?
```

File: fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/dashboard/CardsRestClient.kt (hunk 1 above)
Comment (antonis) [reply fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/dashboard/CardsRestClient.kt]: Great catch @justtwago 👍 
I accepted the suggestion and changed the CardModel fields to be nullable too with 6a1d32ef84faf54246f5061c093956150e3f5ae9

File: fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/dashboard/CardsRestClient.kt (hunk 2 above)
Comment (antonis) [reply fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/dashboard/CardsRestClient.kt]: As above https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2923#discussion_r1425181603

**Review by justtwago — APPROVED** [review]
Thank you for the change Antonis! Looks perfect to me! 🥇

**Comment by antonis** [comment]
As discussed I'm proceeding with merging to unblock the rest of the development. I don't expect that this PR adds any breakages since it just enhances the expected response.
We can revisit if there are BE changes from the spec in a follow up PR.

**Pull request merged**